test/widgets/trip_page_builder_test.dart

```diff
@@ -0,0 +1,270 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:motis_mitfahr_app/account/models/profile.dart';
+import 'package:motis_mitfahr_app/drives/pages/drives_page.dart';
+import 'package:motis_mitfahr_app/rides/pages/rides_page.dart';
+import 'package:motis_mitfahr_app/util/supabase.dart';
+import 'package:motis_mitfahr_app/util/trip/drive_card.dart';
+import 'package:motis_mitfahr_app/util/trip/ride_card.dart';
+
+import '../util/factories/drive_factory.dart';
+import '../util/factories/model_factory.dart';
+import '../util/factories/profile_factory.dart';
+import '../util/factories/ride_factory.dart';
+import '../util/mock_server.dart';
+import '../util/pump_material.dart';
+import '../util/request_processor.dart';
+import '../util/request_processor.mocks.dart';
+
+void main() {
+  late List<Map<String, dynamic>> drives;
+  late Profile profile;
+  final MockRequestProcessor processor = MockRequestProcessor();
+
+  setUpAll(() async {
+    MockServer.setProcessor(processor);
+  });
+  setUp(() async {
+    reset(processor);
+    profile = ProfileFactory().generateFake(id: 1);
+    SupabaseManager.setCurrentProfile(profile);
+  });
+
+  // since the code for the TripPageBuilder is the same for both pages,
+  // we only test the DrivesPage and assume that the RidesPage works the same.
+  // This test is to show that if the ridesPage is called there are rideCards shown.
+  testWidgets('shows Ride card for RidesPage', (WidgetTester tester) async {
+    final List<Map<String, dynamic>> rides = [
+      RideFactory()
+          .generateFake(
+            id: 1,
+            endTime: DateTime.now().add(const Duration(days: 1)),
+            hideInListView: false,
+            rider: NullableParameter(null),
+            drive: NullableParameter(null),
+            driveId: 3,
+            createDependencies: false,
+          )
+          .toJsonForApi(),
+      RideFactory()
+          .generateFake(
+            id: 2,
+            endTime: DateTime.now().add(const Duration(days: 1)),
+            hideInListView: false,
+            rider: NullableParameter(null),
+            drive: NullableParameter(null),
+            driveId: 4,
+            createDependencies: false,
+          )
+          .toJsonForApi(),
+    ];
+    whenRequest(processor, urlMatcher: equals('/rest/v1/rides?select=%2A&rider_id=eq.1&order=start_time.asc.nullslast'))
+        .thenReturnJson(rides);
+    whenRequest(processor, urlMatcher: matches(RegExp(r'/rest/v1/drives.*id=eq\.3')), methodMatcher: equals('GET'))
+        .thenReturnJson(DriveFactory().generateFake(id: 1).toJsonForApi());
```

Comment:
The id here should probably match up with the prevopis driveId to be a bit more realistic. You can create the drive before the call and then use it throughout the test

Reply:
yes thanks. forgot to do that.

Reply:
yes thanks. forgot to do that.

Reply:
now that i'm thinking why is this relevant? that has nothing to do the the Widgets I'm testing, but with the cards.

Reply:
Well we should return consistent data via the api? Who knows what our code relies on / will rely on.

Reply:
but here I'm returning drives for my ride and in the other cases drives for myself so shouldn't they be different

Reply:
Maybe I should be more clear about what I mean: You are explicitly loading "the drive with id=3" in the request. Then you return a drive with id=1. That's just weird.

Reply:
ah yes sorry my mistake